tests/test_write_preds.py

```diff
@@ -0,0 +1,248 @@
+import csv
+import os
+import shutil
+import tempfile
+import unittest
+import src.tasks.tasks as tasks
+import torch
+from src import evaluate
+import os.path
+from main import evaluate_and_write
+import pandas as pd
+from src.models import MultiTaskModel
+from unittest import mock
+from allennlp.data.token_indexers import SingleIdTokenIndexer
+from allennlp.data import Instance, Token, vocabulary
+from ..allennlp_mods.numeric_field import NumericField
+from allennlp.data.fields import LabelField, ListField, MetadataField, TextField
+
+
+def model_forward(task, batch, predict=True):
+    if task.name == "sts-b":
+        return {"n_exs": 4, "preds": [5.0, 4.0]}
+    return {"n_exs": 4, "preds": [0, 1, 1, 1]}
+
+
+class TestWritePreds(unittest.TestCase):
+    def sentence_to_text_field(self, sent, indexers):
+        """ Helper function to map a sequence of tokens into a sequence of
+        AllenNLP Tokens, then wrap in a TextField with the given indexers """
+        return TextField(list(map(Token, sent)), token_indexers=indexers)
+
+    def setUp(self):
+        """
+        Since we're testing write_preds, we need to mock model predictions and the parts 
+        of the model, arguments, and trainer needed to write to predictions. 
+        Unlike in update_metrics tests, the actual contents of the examples in val_data 
+        is not the most important as long as it adheres to the API necessary for examples
+        of that task. 
+        """
+        self.temp_dir = tempfile.mkdtemp()
+        self.path = os.path.join(self.temp_dir, "temp_dataset.tsv")
+        self.stsb = tasks.STSBTask(self.temp_dir, 100, "sts-b", tokenizer_name="MosesTokenizer")
+        self.wic = tasks.WiCTask(self.temp_dir, 100, "wic", tokenizer_name="MosesTokenizer")
+        stsb_val_preds = pd.DataFrame(
+            data=[
+                {
+                    "idx": 0,
+                    "labels": 5.00,
+                    "preds": 5.00,
+                    "sent1_str": "A man with a hard hat is dancing.",
+                    "sent2_str": "A man wearing a hard hat is dancing",
+                },
+                {
+                    "idx": 0,
+                    "labels": 4.750,
+                    "preds": 0.34,
+                    "sent1_str": "A young child is riding a horse.",
+                    "sent2_str": "A child is riding a horse.",
+                },
+            ]
+        )
+        wic_val_preds = pd.DataFrame(
+            data=[
+                {
+                    "idx": 0,
+                    "sent1": "Room and board. ",
```

Comment:
The spacing and punctuation seem a little all over the place. It probably doesn't matter, but maybe worth cleaning up?

Reply:
e.g. "Hook a fish"  and "He hooked a snake accidentally , ..." below